tests/ssl-opt.sh

```diff
@@ -14457,6 +14457,531 @@ run_test    "TLS 1.2 ClientHello indicating support for deflate compression meth
             -c "Handshake was completed" \
             -s "dumping .client hello, compression. (2 bytes)"
 
+# Handshake defragmentation testing
+
+# To guarantee that the handhake messages are large enough and need to be split
+# into fragments, the tests require certificate authentication. The party in control
+# of the fragmentation operations is OpenSSL and will always use server5.crt (548 Bytes).
+requires_certificate_authentication
+run_test    "Handshake defragmentation on client (no fragmentation, for reference)" \
+            "$O_NEXT_SRV" \
+            "$P_CLI debug_level=4 " \
+            0 \
+            -C "reassembled record" \
+            -C "waiting for more fragments"
+
+requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_3
+requires_certificate_authentication
```

Comment:
Why no dependency on ECDSA here?

Reply:
I commented on the same quesiton on the development https://github.com/Mbed-TLS/mbedtls/pull/9989#discussion_r1973751676

In short its not needed.

Reply:
https://github.com/Mbed-TLS/mbedtls/pull/9989#discussion_r1973751676 doesn't apply to 3.6, which has RSA-DHE.

So what in our test coverage makes something needed in development but not in 3.6?

Reply:
That's not really clear to me either. I can see that it passes but I don't really understand why it's needed in development but not 3.6.

Considering we really want to merge this soon, and the thing we don't understand is just the details of a dependency declaration (and possibly some corner case in our complex and underdocumented autodetection logic), I'm OK approving without understanding this specific point.

But I'd really like us to get to the bottom of this at some point.